### PR TITLE
createISTPattrs needed documentation and testing

### PR DESCRIPTION
**Pull request by balarsen**
- [X ] Pull request has descriptive title
- [ X] Pull request gives overview of changes
- [ X] New code has inline comments where necessary
- [ X fixed] Any new modules, functions or classes have docstrings consistent with SpacePy style
- [ X] Added an entry to CHANGELOG if fixing a major bug or providing a major new feature
- [ X fixed ] New features and bug fixes should have unit tests
- [ X ] Relevant issues are linked to (e.g. `See issue #` or `Closes #`)

This is related to #305, `createISTPattrs` needed else handler for datatype. Documentation was added and unittests, really regression tests as this is a "factory" function

### CODE REVIEW

**Comment by balarsen** [comment]
@jtniehof please look both at the code updates but also that the returned meatadata is compete. Mayeb this should move to `pycdf/istp.py`

**Comment by balarsen** [comment]
Actually hang off. It somehow grabbed other changes in a different branch

**Comment by balarsen** [comment]
Ok, offending commits removed, this is good for review, @jtniehof

File: CHANGELOG [hunk 1]
```diff
@@ -6,6 +6,7 @@ coordinates
  - Include upstream fix for SPH->RLL coordinate transform
 datamodel
  - Fix in SpaceData.toXXX() methods which would fail in some cases, now work as intended
+ - Documentation update and default untis change in createISTPattrs
```
Comment (jtniehof) [inline CHANGELOG:9]:
```suggestion
 - Documentation update and default units change in createISTPattrs
```

File: spacepy/datamodel.py [hunk 2]
```diff
@@ -1834,8 +1834,44 @@ def dmcopy(dobj):
...
         return copy.copy(dobj)
 
-def createISTPattrs(datatype, ndims=1, vartype=None, units=None, NRV=False):
+def createISTPattrs(datatype, ndims=1, vartype=None, units=' ', NRV=False):
     '''Return set of unpopulated attributes for ISTP compliant variable
+
+    Parameters
+    ----------
+    datatype : str
+        datatype of variable to reate metadata for (data|support_data|metadata)
```
Comment (jtniehof) [inline spacepy/datamodel.py:1843]:
```suggestion
    datatype : {'data', 'support_data', 'metadata'}
        datatype of variable to create metadata for.
```

See "parameters" in the [numpydoc docs](https://numpydoc.readthedocs.io/en/latest/format.html#sections)

File: spacepy/datamodel.py [hunk 3]
```diff
@@ -1834,8 +1834,44 @@ def dmcopy(dobj):
...
+    ndims : int
+        number of dimensions, default=1
+    vartype : str
+        The type of the variable (float|char|int|epoch|tt2000), default=float
```
Comment (jtniehof) [inline spacepy/datamodel.py:1847]:
```suggestion
    vartype : {'float', 'char', 'int', 'epoch', 'tt2000'}
        The type of the variable, default=float
```

**Review by jtniehof — CHANGES_REQUESTED** [review]
Long-term I agree this should be pulled into other ISTP handling, but this is nice for now.

Good except typos and such. There are some other things in the numpydoc guide regarding labeling optional stuff; I think it would make more sense to modify this commit or make a patch-up commit rather than trying to apply the suggestions (I think we've had problems with multi-line suggestions in the past.) That would then be an opportunity to apply the optional label per the numpydoc.

**Comment by jtniehof** [comment]
Yikes, those are non-descriptive commit messages...is that what you get when you accept the suggestion in the PR review?

**Comment by balarsen** [comment]
> 
> 
> Yikes, those are non-descriptive commit messages...is that what you get when you accept the suggestion in the PR review?

I guess so, I never saw any messages go by... any ideas on how to modify them?

**Comment by balarsen** [comment]
I cleaned those up, followed https://www.educative.io/edpresso/how-to-change-a-git-commit-message-after-a-push and used `git rebase -i HEAD~3` pretty cool interactive functionality

**Comment by jtniehof** [comment]
Yep, rebase is awesome, makes new commits so have to force push, as you noticed.

**Review by jtniehof — APPROVED** [review]
Looks good here. Thanks.

**Pull request merged**